### DISCUSSION TITLE
Why can't I bind parent value in TemplatedControl?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12077] [Q&A] [answered]
I created a TemplatedControl:
XAML:
```
<Styles xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:controls="using:AvaloniaApplication1">
  <Design.PreviewWith>
    <controls:TemplatedControl1 />
  </Design.PreviewWith>

  <Style Selector="controls|TemplatedControl1">
    <!-- Set Defaults -->
    <Setter Property="Template">
      <ControlTemplate>
        <TextBlock IsVisible="{Binding $parent[controls:TemplatedControl1].IsChecked}" Text="Templated Control" />
      </ControlTemplate>
    </Setter>
  </Style>
</Styles>
```
Backend:
```
using Avalonia.Controls.Primitives;

namespace AvaloniaApplication1;

public class TemplatedControl1 : ToggleButton
{
}
```

When the program run, it reports this error on  `IsVisible="{Binding $parent[controls:TemplatedControl1].IsChecked}"`

> System.ArgumentException: 'Unable to resolve type controls:TemplatedControl1 from any of the following locations: '

What's wrong with it?

**Comment by timunie** [comment]
Try `IsVisible="{TemplateBinding IsChecked}"` or `IsVisible="{Binding IsChecked, RelativeSource={ReleativeSoruce TemplatedParent}}"`

See also: https://docs.avaloniaui.net/docs/next/guides/custom-controls/how-to-create-templated-controls#data-binding

**Reply by kyuranger** [reply]
The  `IsVisible="{TemplateBinding IsChecked}"`works. 
Whereas, when I want to use the not operator like below:
`IsVisible="{TemplateBinding !IsChecked}"`
it reports error.

> Avalonia.Data.Core.ExpressionParseException: Unexpected '!'.


In addition,
`IsVisible="{Binding IsChecked, RelativeSource={ReleativeSoruce TemplatedParent}}"`
no works with an error:

> Unable to find suitable setter or adder for property RelativeSource of type Avalonia.Markup.Xaml:Avalonia.Markup.Xaml.MarkupExtensions.ReflectionBindingExtension for argument Avalonia.Markup:Avalonia.Data.Binding

How can I solve this?

**Reply by timunie** [reply, marked as answer]
looks like a typo I had above. `IsVisible="{Binding IsChecked, RelativeSource={RelativeSoruce TemplatedParent}}"`

**Reply by kyuranger** [reply]
Well, it works and it works with non-operator also. Thank you.

By the way, I'm not sure, but is it a bug in Avalonia that `$parent[controls:TemplatedControl1]` inside  TemplatedControl doesn't work?

**Reply by jp2masa** [reply]
`$parent` works on logical tree and `RelativeSource` on visual tree. There is a warning at the end of this docs section: https://docs.avaloniaui.net/docs/data-binding/binding-to-controls#binding-to-an-ancestor.
